### ISSUE TITLE
TagHelper not recognized until recompilation

### ISSUE THREAD

**Issue by stil** [aspnet/AspNetCore#7880]
### Describe the bug
Tag helper is not processed unless I edit cshtml and make a page refresh.

### To Reproduce
1. Using this version of ASP.NET Core v2.2.0, VS 2019 Preview 3
2. Create empty ASP.NET Core solution.
3. Add tag helper

```csharp
using Microsoft.AspNetCore.Razor.TagHelpers;

namespace WebApplication1.TagHelpers
{
    public class ExampleTagHelper : TagHelper
    {
        public override void Process(TagHelperContext context, TagHelperOutput output)
        {
            output.TagName = "input";
        }
    }
}
```

Add to ViewImports

```cshtml
@using WebApplication1
@using WebApplication1.Models
@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
@addTagHelper *, WebApplication1
```

Add to `Home/Index.cshtml`

```cshtml
@{
    ViewData["Title"] = "Home Page";
}

<div class="text-center">
    <h1 class="display-4">Welcome</h1>
    <p>Learn about <a href="https://docs.microsoft.com/aspnet/core">building Web apps with ASP.NET Core</a>.</p>
</div>

<example></example> 
```

### Expected behavior
Tag helper should become an input tag. Instead it is unchanged unless I edit cshtml code, for example adding a space, save it and do a page refresh.

### Screenshots
![taghelpers](https://user-images.githubusercontent.com/714004/53275932-560f9b80-36fd-11e9-857c-1b5969a36a05.gif)

**Comment by pranavkm** [comment]
Likely a dup of https://github.com/aspnet/AspNetCore/issues/6825. This is fixed in 3.0-preview3 release. In the meanwhile, I'd recommend staying on the 2.2 SDK (using a global.json to specify an exact version) until we get a resolution for this.

**Comment by stil** [comment]
I can confirm it works with 2.2 SDK
